src/dmd/mars.d

```diff
@@ -2671,6 +2715,9 @@ private void reconcileCommands(ref Param params, size_t numSrcFiles)
         }
     }
 
+    if (params.makeDeps && !params.oneobj)
```

Comment:
I think this is unnecessarily restrictive: anything that builds to a single output file should be fine, e.g. a library.
BTW: just checking `params.makeDeps` later is enough and it would make it easier to extend to other cases.

Reply:
> anything that builds to a single output file should be fine

This is why I disabled makedeps if `oneobj` is false, but yes it would be better to test for `multiobj` instead.
I saw in the logic in the same function that `oneobj` is set as soon as there is a link operation.
Only for static lib I believe it can be unset.

Reply:
These `oneobj` and `multiobj` flags sometimes only describe intermediate steps (e.g. when building a library, `multiobj` is used, but the object files are not actually written to disk). So a slightly more elaborate interpretation of the `params` might be necessary to support other cases.

The generated object file might also be an intermediate file, e.g. when automatically linking to an executable. I'd expect the executable to be the target of the make dependency. Actually, all generated files could be listed there. For example, if the build system also deals with generated documentation. That could also cover building multiple object files wih `-c`.

No immediate action required though, this PR was already a good step forward.